### PR TITLE
DYN-6535 DynamoRevit Improve Search Fix 2

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Fixing specific case in DynamoRevit Search algorithm
The specific case "element set parameter" was failing on Revit 2025 (not finding anything), then I've modified the code so that check if the first term is a category if that is the case it will create a specific query using the first term as category and the next terms as node name.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fixing specific case in DynamoRevit Search algorithm

### Reviewers

@QilongTang 

### FYIs

@avidit

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
GIF showing the behavior in Revit 2025
chrome_CMhi2PDSG5

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs [hunk 1]
```diff
@@ -335,6 +349,33 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm)
...
+
+                //This will only valid when the search term has empty spaces
+                if (hasEmptySpaces)
+                {
+                    //Check if the first term of the search criteria match any category
+                    var possibleCategory = searchTerm.Split(' ')[0];
+                    firstTermIsCategory = dynamoModel.SearchModel.Entries.Where(entry => IsMatchingCategory(possibleCategory, entry.FullCategoryName) == true && !string.IsNullOrEmpty(possibleCategory)).Any();
+
+                    //Get the node matching the Category provided in the search term
+                    var matchingCategory = dynamoModel.SearchModel.Entries.Where(entry => IsMatchingCategory(possibleCategory, entry.FullCategoryName) == true && !string.IsNullOrEmpty(possibleCategory)).FirstOrDefault();
```
Comment (reddyashish) [inline src/DynamoCore/Utilities/LuceneSearchUtility.cs:362]: The same Where function on Entries is running twice, store the result and use it again.

Comment (reddyashish) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:362]: Also can we check if the possibleCategory is null or empty before the where call.

Comment (RobertGlobant20) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs]: Fixed in the next commit:
https://github.com/DynamoDS/Dynamo/pull/14917/commits/c62ba640baa2c7623d3711f74f27c6ad9f227c47

**Comment by QilongTang** [comment]
Also @RobertGlobant20 Now that the API breaking change has been fixed, you should be able to test this with master changed pulled in

**Comment by RobertGlobant20** [comment]
> Also @RobertGlobant20 Now that the API breaking change has been fixed, you should be able to test this with master changed pulled in

@QilongTang I just updated the master branch with latest changes and merge it into the feature branch and is working as expected.
chrome_4oThzDcwFn

**Comment by QilongTang** [comment]
@RobertGlobant20 It's strange that the git diff now shows you touched all those file of node help docs..

**Review by QilongTang — COMMENTED** [review]
LGTM, is there a way to add a unit test for this improvement?

**Comment by RobertGlobant20** [comment]
> @RobertGlobant20 It's strange that the git diff now shows you touched all those file of node help docs..

@QilongTang fixed, When merging the master branch I did a "Squash and Merge"  but I just reverted and did a "Create a merge commit" and now is reflecting the right files.

**Comment by RobertGlobant20** [comment]
> LGTM, is there a way to add a unit test for this improvement?

@QilongTang probably just for DynamoSandbox specifically testing that when you type "list " or "input " you should get nodes under that specific category.

**Comment by QilongTang** [comment]
> > LGTM, is there a way to add a unit test for this improvement?
> 
> @QilongTang probably just for DynamoSandbox specifically testing that when you type "list " or "input " you should get nodes under that specific category.

That's fine, can you give it a try to wrap this up?

**Comment by RobertGlobant20** [comment]
> > > LGTM, is there a way to add a unit test for this improvement?
> > 
> > 
> > @QilongTang probably just for DynamoSandbox specifically testing that when you type "list " or "input " you should get nodes under that specific category.
> 
> That's fine, can you give it a try to wrap this up?

@QilongTang I just tested several criteria in DynamoSandbox and got the same results than previous versions.
This change just apply for a specific case in Revit due that when you search  for "element set parameter" Revit has several nodes which name contains "set" and "parameters" so was showing those nodes first and after my fix is showing the "Element.Setparameterbyname" will be shown at the top and then the other nodes.
Then I think we cannot create a unit test  for this feature unless all the revit nodes are indexed.

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@RobertGlobant20 Thanks. LGTM then

**Pull request merged**